pkg/apis/triggers/v1alpha1/interceptor_types.go

```diff
@@ -14,6 +14,8 @@ type InterceptorInterface interface {
 	// see if processing should continue and InterceptorResponse.Status.Code to distinguish between the kinds of errors
 	// (i.e user errors vs system errors)
 	Process(ctx context.Context, r *InterceptorRequest) *InterceptorResponse
+
+	InitializeContext(ctx context.Context)
```

Comment:
What do you think of changing each interceptor so that in the process func itself we do something like this: `logger := logging.FromContext(ctx)`


The idea is that the context might contain request specific info (e.g. eventID/requestID).

Reply:
that makes sense, ill update the interceptors and the documentation around how this should work.

do we do the same thing with the kube client? then we can just remove this function altogether from the interface

Reply:
yup! exactly! 

https://github.com/tektoncd/triggers/blob/4939226120cdf4646e1f5941fb1b10b8f866abac/vendor/knative.dev/pkg/client/injection/kube/client/client.go